command/host_create.go

```diff
@@ -33,11 +34,14 @@ func (c *createHost) ParseParams(params map[string]interface{}) error {
 		c.CreateHost.Background = true
 	}
 
-	// if a filename is defined, get the params from the file
 	if _, ok := params["file"]; ok {
 		fileName := fmt.Sprintf("%v", params["file"])
-		return errors.Wrapf(utility.ReadYAMLFile(fileName, &c.CreateHost),
-			"error reading from file '%s'", fileName)
+		c.File = fileName
+		fmt.Println(len(params))
+		if len(params) > 1 {
+			return errors.New("Programmer error: no params should be defined when using a file to parse params")
+		}
+		return nil
 	}
 
 	decoder, err := mapstructure.NewDecoder(&mapstructure.DecoderConfig{
```

Comment:
if a file was passed should we skip reading in the rest of the params?

Reply:
yes. In my most recent change I added return nil to the end of the if statement.

Reply:
I missed that :-)